### PR TITLE
feat: add support for exporting dependency graph

### PR DESCRIPTION
**Pull request by smira**
This graph shows relationship between resources and controllers, which
might be invaluable for debugging and understanding system dependencies.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Review by andrewrynhard — DISMISSED** [review]

File: pkg/controller/dependency.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+// DependencyEdgeType is edge type in controller graph.
+type DependencyEdgeType int
+
+// Controller graph edge types.
+const (
+	EdgeManages DependencyEdgeType = iota
+	EdgeDependsHard
+	EdgeDependsWeak
```
Comment (andrewrynhard) [inline pkg/controller/dependency.go:21]: Not for this PR but I missed this nit the other day. Strong and Weak go better together naturally. Hard and Soft likewise. I know this aligns with the current convention, but just my 2 cents for the future.

Comment (smira) [reply pkg/controller/dependency.go:21]: thanks, let me update it for this PR until it spreads too far!

**Review by andrewrynhard — DISMISSED** [review]

**Pull request merged**